### DISCUSSION TITLE
Merge objects together.

### DISCUSSION THREAD

**Discussion by raphaelsty** [online-ml/river#1017] [Ideas]
Hi,

This discussion to ask if it would be relevant to be able to merge two river objects.

I see a use case in the context of running several independent processes. A user might want to merge the objects at regular intervals. I'm not sure if it's relevant or if it fits with the river philosophy. It is obvious to implement for the mean statistic but not sure if it is simple for other methods like linear models.

```python
from river import stats

a = stats.Mean()
a.update(1)

b = stats.Mean()
b.update(4)

c = a.join(b)
c.get()
# 2.5
```

Maybe the question has already been asked :)

**Comment by MaxHalford** [comment]
Thanks to our friend from Brazil @smastelini we can do this:

```py
from river import stats

a = stats.Mean()
a.update(1)

b = stats.Mean()
b.update(4)

c = a + b
c.get()
```

:)

**Reply by MaxHalford** [reply]
We should document this feature really.

**Reply by smastelini** [reply]
I totally agree! It is on my to-do list to create a guide for this kind of operation and also another one for `warm_up_mode` and `pure_inference_mode`.

About merging objects, I think with some reading about federated learning, we could make stuff like:

```py
lm_a = linear_model.LinearRegression()
lm_b = linear_model.LinearRegression()
# lm_a and lm_b are trained with different samples of a stream
# (maybe in different nodes in a distributed computing setting)
```
and then

```py
lm_c = lm_a + lm_b
```

And we have a new model whose weights are a combination of the ones in models A and B.

Sounds like the future, huh?

**Reply by raphaelsty** [reply]
Amazing! Of course it's the future! 🥳
